parity-clib/examples/cpp/main.cpp

```diff
@@ -26,30 +26,33 @@
 #include <thread>
 namespace {
 using namespace std::literals::string_literals;
-using parity::ethereum::ParityEthereum;
 using parity::ethereum::parity_subscription;
+using parity::ethereum::ParityEthereum;
 
 void parity_subscribe_to_websocket(ParityEthereum &ethereum);
 void parity_rpc_queries(ParityEthereum &);
 parity::ethereum::ParityEthereum parity_run(const std::vector<std::string> &);
 
-const int SUBSCRIPTION_ID_LEN = 18;
-const size_t TIMEOUT_ONE_MIN_AS_MILLIS = 60 * 1000;
-const unsigned int CALLBACK_RPC = 1;
-const unsigned int CALLBACK_WS = 2;
+constexpr uint32_t SUBSCRIPTION_ID_LEN = 18;
+constexpr uint32_t TIMEOUT_ONE_MIN_AS_MILLIS = 60 * 1000;
+enum class parity_callback_type : size_t {
+   CALLBACK_RPC = 1,
+   CALLBACK_WS = 2,
+};
 
 struct Callback {
-  size_t type;
+  parity_callback_type type;
```

Comment:
generally I try to avoid variable declarations without initializers. For this I especially like the newer `auto` based initialization scheme so that you cannot forget initializing a variable:

For example:
```
auto type = parity_callback_type{ /* however it needs to be initialized */ };
```